### DISCUSSION TITLE
fpp Build Step Failure

### DISCUSSION THREAD

**Discussion by MichaelStarks** [nasa/fprime#1205] [Q&A] [answered]
Hello,

I am trying to follow the build instructions on the repo, but run into a cmake error. I think the version of Scala-sbt is too new, the `in` keyword has been deprecated as of version 1.5.0.

```
  /home/pi/f_prime/fprime/build-fprime-automatic-native/fpp-tools/src/compiler/build.sbt:13:
  warning: method in in trait ScopingSetting is deprecated (since 1.5.0):
  `in` is deprecated; migrate to slash syntax -
  https://www.scala-sbt.org/1.x/docs/Migrating-from-sbt-013x.html#slash

    testOptions in Test += Tests.Argument(TestFrameworks.ScalaTest, "-oNCXELOPQRM"),
                ^

  
  /home/pi/f_prime/fprime/build-fprime-automatic-native/fpp-tools/src/compiler/build.sbt:4:
  warning: method in in trait ScopingSetting is deprecated (since 1.5.0):
  `in` is deprecated; migrate to slash syntax -
  https://www.scala-sbt.org/1.x/docs/Migrating-from-sbt-013x.html#slash

  organization in ThisBuild := "gov.nasa.jpl"

               ^

  
  /home/pi/f_prime/fprime/build-fprime-automatic-native/fpp-tools/src/compiler/build.sbt:5:
  warning: method in in trait ScopingSetting is deprecated (since 1.5.0):
  `in` is deprecated; migrate to slash syntax -
  https://www.scala-sbt.org/1.x/docs/Migrating-from-sbt-013x.html#slash

  scalaVersion in ThisBuild := "2.13.6"

               ^

  make[2]: *** [CMakeFiles/fpp-tools.dir/build.make:75:
  fpp-tools-prefix/src/fpp-tools-stamp/fpp-tools-install] Error 1

  make[1]: *** [CMakeFiles/Makefile2:76: CMakeFiles/fpp-tools.dir/all] Error
  2

  make: *** [Makefile:84: all] Error 2

Call Stack (most recent call first):
  cmake/required.cmake:20 (include)
  cmake/FPrime.cmake:26 (include)
  CMakeLists.txt:12 (include)


[ERROR] CMake erred with return code 1. Partial build-cache remains. Run purge to clean up.

```

Above is the error message that I receive. It appears that I need to downgrade Scala-sbt to before 1.5.x but it is automatically installed during fprime-utils generate. Should I uninstall the current version and reinstall an earlier version before rerunning fprime-utils generate? 

Thank you

**Comment by LeStarch** [comment, marked as answer]
@bocchino can you speak to this?

**Reply by bocchino** [reply]
What version of sbt are you using? I'm using sbt 1.5.5, and I don't see that warning.

As a workaround, you can install v1.0.0 of FPP manually, as described here: https://github.com/fprime-community/fpp/tree/main/compiler. If the correct version of FPP is in your PATH, then the F Prime build will skip the FPP install step.

**Reply by bocchino** [reply]
It looks like we should fix that deprecated syntax. I opened an issue here: https://github.com/fprime-community/fpp/issues/107.

**Reply by MichaelStarks** [reply]
I am also using version 1.5.5. I will try the manual install.

Thank you

**Reply by MichaelStarks** [reply]
The manual install worked.

**Reply by bocchino** [reply]
Great!
